url.bs

```diff
@@ -88,6 +88,256 @@ valid input. User agents, especially conformance checkers, are encouraged to rep
+  <!-- host parser -->
+  <tr>
+   <td><dfn>IPv6-unclosed</dfn>
+   <td>
+    <p>An <a for=/>IPv6 address</a> is missing the closing U+005D (]).
+    <p class=example id=example-ipv6-unclosed"<code>https://[::1</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>domain-invalid-code-point</dfn>
+   <td>
+    <p>The input's <a for=/>host</a> contains a <a>forbidden domain code point</a>.
+    <div class=example id=example-domain-invalid-code-point>
+     <p>Hosts are <a for=string>percent-decoded</a> before being processed when the URL
+     <a>is special</a>, which would result in the following host portion becoming
+     "<code>exa#mple.org</code>".
+     <p>"<code>https://exa%23mple.org</code>"
+    </div>
+   <td>Yes
+  <!-- IPv4 parser -->
+  <tr>
+   <td><dfn>IPv4-empty-part</dfn>
+   <td>
+    <p>An <a for=/>IPv4 address</a> ends with a U+002E (.).
+    <p class=example id=example-ipv4-empty-part>"<code>https://127.0.0.1./</code>"
+   <td>·
+  <tr>
+   <td><dfn>IPv4-too-many-parts</dfn>
+   <td>
+    <p>An <a for=/>IPv4 address</a> has to consist of 4 parts.
+    <p class=example id=example-ipv4-too-many-parts>"<code>https://1.2.3.4.5/</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>IPv4-non-numeric-part</dfn>
+   <td>
+    <p>An <a for=/>IPv4 address</a> part was not numeric.
+    <p class=example id=example-ipv4-non-numeric-part>"<code>https://test.42</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>IPv4-non-decimal-part</dfn>
+   <td>
+    <p>The <a for=/>IPv4 address</a> contains numbers expressed using hexadecimal or octal digits.
+    <p class=example id=example-ipv4-non-decimal-part>"<code>https://127.0.0x0.1</code>"
+   <td>·
+  <tr>
+   <td><dfn>IPv4-out-of-range-part</dfn>
+   <td>
+    <p>An <a for=/>IPv4 address</a> part exceeds 255.
+    <p class=example id=example-ipv4-out-of-range-part>"<code>https://255.255.4000.1</code>"
+   <td>Yes (only if applicable to the last part)
+  <!-- IPv6 parser -->
+  <tr>
+   <td><dfn>IPv6-invalid-compression</dfn>
+   <td>
+    <p>An <a for=/>IPv6 address</a> begins with improper compression.
+    <p class=example id=example-ipv6-invalid-compression>"<code>https://[:1]</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>IPv6-too-many-pieces</dfn>
+   <td>
+    <p>An <a for=/>IPv6 address</a> contains more than 8 pieces.
+    <p class=example id=example-ipv6-too-many-pieces>"<code>https://[1:2:3:4:5:6:7:8:9]</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>IPv6-multiple-compression</dfn>
+   <td>
+    <p>An <a for=/>IPv6 address</a> is compressed in more than one spot.
+    <p class=example id=example-ipv6-multiple-compression>"<code>https://[1::1::1]</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>IPv4-in-IPv6-too-many-pieces</dfn>
+   <td>
+    <p>An <a for=/>IPv6 address</a> with <a for=/>IPv4 address</a> syntax: the IPv6 address has more
+    than 6 pieces.
+    <p class=example id=example-ipv4-in-ipv6-too-many-pieces>"<code>https://[1:1:1:1:1:1:1:127.0.0.1]</code>"
+   <td>Yes
+  <tr>
+   <td><dfn>IPv4-in-IPv6-invalid-code-point</dfn>
```

Comment:
Really seems like this should be split into multiple errors, but I guess that could be refactored later.

Reply:
I think I ended up grouping some obscure ones into this one as there was no great logical separation. And IPv4-in-IPv6 is already pretty obscure so I'm not sure how detailed we really want to go.